.github/scripts/check.sh

```diff
@@ -1,24 +1,70 @@
 #!/usr/bin/env bash
 set -o pipefail
-REGISTRY_BASE_URL="${REGISTRY_BASE_URL:-https://registry.coder.com}"
 set -u
 
-if [[ -n "${VERBOSE:-}" ]]; then
-    set -x
-fi
+REGISTRY_BASE_URL="${REGISTRY_BASE_URL:-https://registry.coder.com}"
 
 status=0
 declare -a modules=()
 declare -a failures=()
+
+# Collect all module directories containing a main.tf file
 for path in $(find . -not -path '*/.*' -type f -name main.tf -maxdepth 2 | cut -d '/' -f 2 | sort -u); do
     modules+=("${path}")
 done
+
 echo "Checking modules: ${modules[*]}"
+
+# Function to update the component status on Instatus
+update_component_status() {
+    local component_status=$1
+    # see https://instatus.com/help/api/components
+    (curl -X PUT "https://api.instatus.com/v1/$INSTATUS_PAGE_ID/components/$INSTATUS_COMPONENT_ID" \
+        -H "Authorization: Bearer $INSTATUS_API_KEY" \
+        -H "Content-Type: application/json" \
+        -d "{\"status\": \"$component_status\"}")
+}
```

Comment:
Can we check for the required InStatus environment variables being set upfront and exit with a non-zero exit status if not provided? Right now we will exit with

```
./.github/scripts/check.sh: line 22: INSTATUS_PAGE_ID: unbound variable

```